src/antelope/stores/contract.ts

```diff
@@ -13,27 +13,146 @@
+import { AntelopeError, EvmContractFactoryData, EvmTransaction } from 'src/antelope/types';
+import { ethers } from 'ethers';
 
 
-export interface ContractState {
-    abi: string;
+export interface ContractStoreState {
+    factory: EvmContractFactory;
+    cachedContracts: Record<string, EvmContract>
+    processing: string[];
 }
 
 const store_name = 'contract';
 
 export const useContractStore = defineStore(store_name, {
-    state: (): ContractState => (contractInitialState),
+    state: (): ContractStoreState => (contractInitialState),
     getters: {
     },
     actions: {
         trace: createTraceFunction(store_name),
-        init: createInitFunction(store_name),
+        init() {
+            useFeedbackStore().setDebug(store_name, isTracingAll());
+        },
+
+        async getContract(address: string): Promise<EvmContract | null> {
+            const addressLower = address.toLowerCase();
+
+            if (typeof this.cachedContracts[addressLower] !== 'undefined') {
+                return this.cachedContracts[addressLower];
+            }
+
+            if (this.processing.includes(addressLower)) {
+                await new Promise(resolve => setTimeout(resolve, 300));
+                return this.getContract(address);
+            }
+            this.processing.push(addressLower);
+
+            const index = this.processing.indexOf(addressLower);
+            let contract = { address: address };
+            try {
+                const indexer = (useChainStore().loggedChain.settings as EVMChainSettings).getIndexer();
+                const response = await indexer.get(`/contract/${address}?full=true&includeAbi=true`);
+
+                if (response.data?.success && response.data.results.length > 0){
+                    contract = response.data.results[0];
+                }
+            } catch (e) {
+                console.warn(`Could not retrieve contract ${address}: ${e}`);
+                this.processing.splice(index, 1);
+                return null;
+            }
+            this.addContractToCache(address, contract);
+            this.processing.splice(index, 1);
+            return this.$state.factory.buildContract(contract);
+        },
+
+        addContractToCache(address: string, contractData: EvmContractFactoryData): void {
+            if (!address) {
+                throw new AntelopeError('antelope.contracts.address_required');
+            }
+            const index = address.toString().toLowerCase();
+            const contract = this.factory.buildContract(contractData);
+
+            if(
+                typeof this.cachedContracts[index] === 'undefined'
+                || (contract.abi ?? []).length > 0 && !this.cachedContracts[index].abi
+                || (contract.abi ?? []).length > 0 &&(contract.abi ?? []).length > (this.cachedContracts[index].abi?.length ?? 0)
+            ){
+                this.cachedContracts[index] = contract;
+            }
+        },
```

Comment:
Would you mind pushing this function to the bottom of the actions section?
This way we can group those functions that actually commit some data into the store.
Check Balances and Account stores. They are following this convention to quickly identify any function that is actually performing as a commit in the store.


![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/171a95d9-c69a-40b8-b85e-8a34a5d97898)

![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/00b053a6-278c-4ba1-8e02-07a95e1cb17c)

They are at the bottom of the list of functions under the // commits comment to easily identify them.